snuba/web/rpc/v1/resolvers/common/aggregation.py

```diff
@@ -239,6 +249,75 @@ def from_alias(alias: str) -> "CustomColumnInformation":
         return CustomColumnInformation(column_type, referenced_column, metadata)
 
 
+def convert_to_conditional_aggregation(
+    in_msg: TraceItemTableRequest | TimeSeriesRequest,
+) -> None:
+    """
+    Up to this point we support aggregation, but now we want to support conditional aggregation, which only aggregates
+    if the field satisfies the condition: https://clickhouse.com/docs/en/sql-reference/aggregate-functions/combinators#-if
+
+    For messages that don't have conditional aggregation, this function replaces the aggregation with a conditional aggregation,
+    where the filter is null, and every field is the same. This allows code elsewhere to set the default condition to always
+    be true.
+
+    The reason we do this "transformation" is to avoid code fragmentation down the line, where we constantly have to check
+    if the request contains `AttributeAggregation` or `AttributeConditionalAggregation`
+    """
+
+    def _add_conditional_aggregation(
```

Comment:
If I'm following this correctly, this function does two things:
1. It traverses the tree represented by `TraceItemTableRequest` and `TimeSeriesRequest`.
2. When it finds an aggregation, it converts that aggregation to a conditional one.

I suspect the code would be easier to read if we separated these two mostly unrelated concerns. The visitor pattern is one way to achieve this (I'm happy to explain this if you're unfamiliar with the visitor pattern idea).